src/preact/base-element.js

```diff
@@ -684,17 +684,22 @@ export class PreactBaseElement extends BaseElement {
     // Add AmpContext with renderable/playable proeprties.
     const v = <WithAmpContext {...this.context_}>{comp}</WithAmpContext>;
 
-    if (this.hydrationPending_) {
-      this.hydrationPending_ = false;
-      hydrate(v, this.container_);
-    } else {
-      const replacement = lightDomTag
-        ? childElementByAttr(this.container_, RENDERED_ATTR)
-        : null;
-      if (replacement) {
-        replacement[RENDERED_PROP] = true;
+    try {
```

Comment:
What is this try/catch added for? Do we have a test to make sure this is added behavior we're looking for.

Reply:
Here's a justification for this change...
- The `catch(err)` here  always re-throws the error, so there's no actual changes to this code's "flow"
- This is catching any rendering errors ... for example, in `BentoAppBanner`, there's something like `userAssert(props.id, 'component requires an id')` ... so this `catch` is designed to catch these rendering assertions
- The purpose of the `catch` is to reject the `renderDeferred_` promise.  
  - WITHOUT this change, the promise would NEVER resolve/reject ... it would just hang, forever, and the component would never render. In tests, they would just hang.  
  - WITH this change, the tests can properly assert that the component fails to render.  

## Risks with this change:
- Rendering errors in a Bento component would no longer "silently fail" ...

Ideally, the Bento framework would have error-handling higher up (I have no idea if it does).  If so, this change would "fix" the bug that this error handling was unreachable.

Worst case, though, if there is no higher-up error handling, this change _could_ trigger an `unhandledrejection` error on the page. 
IMO, this is actually desired, since a render error should certainly be caught and handled somehow.

What do you think?

Reply:
Agreed, this is fine but would be good to separate out.

Reply:
This change was actually necessary to be able to write unit tests for rendering issues.  WIthout the rejected promise, the unit test would hang indefinitely; with it, I can properly test; like in `test-amp-app-banner`:
```
      await expect(
        mountElement(html`<amp-app-banner></amp-app-banner>`)
      ).to.be.rejectedWith(/bento-app-banner should have an id/);
```